src/index.ts

```diff
@@ -1,94 +1,116 @@
-  excludeDomains?: string[]; // Exclude results from these domains. Example: ["excludedomain.com", "excludeme.net"]
-  startCrawlDate?: string; // The optional start date (inclusive) for the crawled data. Must be specified in ISO 8601 format. Example: "2023-01-01"
-  endCrawlDate?: string; // The optional end date (inclusive) for the crawled data. Must be specified in ISO 8601 format. Example: "2023-12-31"
-  startPublishedDate?: string; // The optional start date (inclusive) for the published data. Must be specified in ISO 8601 format. Example: "2023-01-01"
-  endPublishedDate?: string; // The optional end date (inclusive) for the published data. Must be specified in ISO 8601 format. Example: "2023-12-31"
+	numResults?: number; // Number of search results to return. Maximum 100. Default 10
+	includeDomains?: string[]; // Include results only from these domains. Example: ["example.com", "sample.net"]
+	excludeDomains?: string[]; // Exclude results from these domains. Example: ["excludedomain.com", "excludeme.net"]
+	startCrawlDate?: string; // The optional start date (inclusive) for the crawled data. Must be specified in ISO 8601 format. Example: "2023-01-01"
+	endCrawlDate?: string; // The optional end date (inclusive) for the crawled data. Must be specified in ISO 8601 format. Example: "2023-12-31"
+	startPublishedDate?: string; // The optional start date (inclusive) for the published data. Must be specified in ISO 8601 format. Example: "2023-01-01"
+	endPublishedDate?: string; // The optional end date (inclusive) for the published data. Must be specified in ISO 8601 format. Example: "2023-12-31"
 }
 
 // The DocumentContent interface represents the content of a document from the /contents endpoint.
 export interface DocumentContent {
-  id: string // The ID of the document.
-  url: string; // The URL of the document.
-  title: string; // The title of the document.
-  extract: string; // The first 1000 tokens of content in the document.
+	id: string; // The ID of the document.
+	url: string; // The URL of the document.
+	title: string; // The title of the document.
+	extract: string; // The first 1000 tokens of content in the document.
 }
 
 // The GetContentsResponse interface represents the response from the /contents endpoint.
 // It includes an array of document content objects.
 export interface GetContentsResponse {
-  contents: DocumentContent[];
+	contents: DocumentContent[];
 }
 
 // The Metaphor class encapsulates the API's endpoints.
 export default class Metaphor {
-  private client: AxiosInstance;
+	private baseURL: string;
+	private headers: Headers;
+
+	constructor(
+		apiKey: string,
+		baseURL: string = "https://api.metaphor.systems"
+	) {
+		this.baseURL = baseURL;
+		this.headers = new Headers({
+			"x-api-key": apiKey,
+			"Content-Type": "application/json",
+		});
+	}
+
+	private async request(
+		endpoint: string,
+		method: string,
+		body?: any
+	): Promise<any> {
```

Comment:
we will need these things to be typed, even if we don't have the axios level typings